### ISSUE TITLE
Uncaught TypeError: promise.then is not a function

### ISSUE THREAD

**Issue by trumb1mj** [danmcclain/ember-deferred-content#6]
I'm trying to use this with my model hook that's returning a Ember.RSVP.hash and I'm getting:  `Uncaught TypeError: promise.then is not a function`

Ember              : 2.9.0
Ember Data     : 2.9.0
jQuery             : 3.1.1

Is this intended to work with models?

**Comment by danmcclain** [comment]
It should work with models, could you wire up an example where this fails?

**Comment by pcambra** [comment]
I'm finding the same issue, a simple model with no relationships or links declared coming out of jsonapi, both with model and RSVP.hash, I'm getting "deferred-content.js:22 Uncaught TypeError: promise.then is not a function(…)", not sure how to build an example, though.

**Comment by alexander-alvarez** [comment]
> I'm trying to use this with my model hook

your model hook is synchronous -- so when you get to your template your model would have been resolved (be the actual data, not a promise).

Isn't the intent of this addon to be used with promises and the such that haven't resolved yet?

```
init() {
 this.set('someValue', this.get('store').find('some-model'))
}

//
{{#deferred-content someValue as |d|}}
...
{{/defered-content}}
```

**Comment by anydasa** [comment]
I have same problem 

![0b91d0a946d02e7d7eae5a9c8fad52b4](https://cloud.githubusercontent.com/assets/7040531/21733132/8c7fb1e8-d465-11e6-9b74-4186488042a8.png)


![c1423f399eaf4682b4f338c7e03c45dd](https://cloud.githubusercontent.com/assets/7040531/21733063/42460a50-d465-11e6-9073-f6bdd65ca3c8.png)

**Comment by allthesignals** [comment]
@anydasa I get the same thing. Returning an ES6 promise though.

**Comment by danmcclain** [comment]
Think this is handled by #9 (which resolved #8), where it will assert that you are passing a promise to the component, please comment if still having the issue, closing for now
